hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileGetInfoHandler.java

```diff
@@ -81,6 +81,8 @@ public void validate(@NonNull final QueryContext context) throws PreCheckExcepti
         }
     }
 
+    // Suppress the warning that we shouldn't throw generic exceptions
+    @SuppressWarnings("java:S112")
```

Comment:
We should use `UncheckedIOException` instead. I wonder, though, why we can have an `IOException` here. We are just reading data from state. 🤔

Reply:
It looks like `upgradeFileStore.getFull()` is throwing **IOException**. Digging deeper collector(of type ByteArrayOutputStream).write is throwing this exception.